### ISSUE TITLE
TFLite convert not working for TF2 models

### ISSUE THREAD

**Issue by alexbarnett12** [Big-Brain-Crew/learn_ml#20]
I trained the generated TF2 model and then used `convert_to_edgetpu.py` to conver the model to TFLite. However, when I ran inference on the coral, it fails with this error:

```
Exception in thread Thread-4:
Traceback (most recent call last):
  File "/usr/lib/python3.7/threading.py", line 917, in _bootstrap_inner
    self.run()
  File "/usr/lib/python3.7/threading.py", line 865, in run
    self._target(*self._args, **self._kwargs)
  File "/home/mendel/learn_ml/coral_inference/classification/base_classifier.py", line 92, in _thread
    for frame in frames_iterator:
  File "/home/mendel/learn_ml/coral_inference/classification/classifier.py", line 97, in frames
    self.interpreter.set_tensor(input_details[0]['index'], input_data)
  File "/home/mendel/.local/lib/python3.7/site-packages/tflite_runtime/interpreter.py", line 399, in set_tensor
    self._interpreter.SetTensor(tensor_index, value)
  File "/home/mendel/.local/lib/python3.7/site-packages/tflite_runtime/interpreter_wrapper.py", line 148, in SetTensor
    return _interpreter_wrapper.InterpreterWrapper_SetTensor(self, i, value)
ValueError: Cannot set tensor: Got value of type NOTYPE but expected type UINT8 for input 0, name: serving_default_flatten_input 
```

It looks like it is expecting a UINT8 input, but somehow the tflite converter was not able to set an input during conversion (hence the NOTYPE). 

Separately, I have successfully converted and ran inference on a TFLite model using this conversion code:

```
converter = tf.lite.TFLiteConverter.from_keras_model(model)
# This enables quantization
converter.optimizations = [tf.lite.Optimize.DEFAULT]
converter.target_spec.supported_types = [tf.int8]
# This ensures that if any ops can't be quantized, the converter throws an error
converter.target_spec.supported_ops = [tf.lite.OpsSet.TFLITE_BUILTINS_INT8]
# And this sets the representative dataset so we can quantize the activations
converter.representative_dataset = representative_dataset_gen_factory("./examples/representative_dataset.npy")
tflite_model = converter.convert()
```

Notably, my code doesn't include 2 lines from Swapnil's conversion script:
```
converter.inference_input_type = tf.uint8
converter.inference_output_type = tf.uint8

```

I believe these lines are causing the problem, though I am not sure how yet.

**Comment by SwapnilPande** [comment]
I've actually encountered this before? What script did you use to build the neural net? Is it train_mnist.py in the root directory?

**Comment by SwapnilPande** [comment]
Can you also point me to the model you are trying to convert?

**Comment by alexbarnett12** [comment]
I used learn_ml/common/train.py, with a minor modification I made locally. If you would like to tinker with this, I can create a PR with a new branch with these changes.

**Comment by alexbarnett12** [comment]
I just tested a few things: removing the two lines about input and output type being uint8 didn't fix the problem, BUT changing the top line to be the tf2 converter did. So it looks like it's some difference within the converters.

**Comment by SwapnilPande** [comment]
I was unable to recreate the issue. I trained using the generated training script, and my app runs inference just fine.

What version of tensorflow are you using?

**Comment by SwapnilPande** [comment]
Closed by #22
